app/src/main/java/com/github/orkest/shazam/ui/ShazamActivity.kt

```diff
@@ -14,24 +14,30 @@ import com.github.orkest.shazam.data.AudioRecognition
 import com.github.orkest.shazam.data.AudioRecording
 import com.github.orkest.shazam.domain.ShazamConstants
 import kotlinx.coroutines.launch
+import java.util.concurrent.CompletableFuture
 
 
-    /**
+/**
      * Composable function for shazaming a song, should be launched in the background of the picture taking activity
      * Records AudioChunks continuously and asks for permission if needed
      */
      @SuppressLint("MissingPermission")
      @Composable
-    fun ShazamSong(activity: Activity) {
+    fun ShazamSong(activity: Activity): CompletableFuture<Song> {
 
         val context = LocalContext.current
         val coroutineScope = rememberCoroutineScope()
+
+        val futureSong = CompletableFuture<Song>()
         LaunchedEffect(Unit) {
             coroutineScope.launch {
                 val permission = ShazamConstants.recordPermissionGranted(context)
                 //Asks for recording permission if not given
-                if (!permission)
+                if (!permission){
                     ShazamConstants.askRecordPermission(activity)
+                futureSong.completeExceptionally(Exception("No permission to record audio"))
+                }
+
```

Comment:
Where do you handle the expection ?

Reply:
It will be handled in the next sprint when combining our code with sara to post a picture with a song